### ISSUE TITLE
Critical Hit system

### ISSUE THREAD

**Issue by MistahStealYoGril** [Anuken/Mindustry-Suggestions#3166]
### Describe the content or mechanics you are proposing.

It would be nice to see some type of critical hit system where if we do get one, one weapon on the enemy gets disabled or they have some type of leakage (oil preferably) that causes them to gradually lose health. 

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

It would probably give players something specific to aim at on a guardian, making it easier or putting them at an advantage over the boss so the player doesn't lose too much of their work. The same applies to multiplayer games.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by MrDuck557** [comment]
isn't this just making it harder for players to storm bases

**Comment by MistahStealYoGril** [comment]
Maybe a little. It is all by chance. So it won't be too difficult if the player is lucky.

**Comment by itcannotbe** [comment]
All I see is someone adding more RNG because...

**Comment by OnyxFinger** [comment]
Mindustry is, as you probably know a game similar to an RTS in some ways (of course it isn't only an RTS), and about consistency.
RNG generally doesn't fit with the RTS style. As for consistency, it doesn't help either.

Remember boiz, critical strike odds can be changed if they're too high, which is why ***I will make sure that the first person saying that "because of how many bullets come out of turrets, most units will be disabled" is sent to Jesus***

**Comment by IxGamerXL** [comment]
_**gets flashbacks of TF2**_

...Yeah, this is very likely not going to work unless you also implement a way for crits to not be random.

Let me add to this: **Weak Points**

Most units have them. Some units don't have a weak point to begin with OR have armored weak points, such as the Oct.

Take for example: the Corvus. It has a large barrel that looks like a mouth. That would be the weak point, thus critical hits for that specific part of the unit.

Some weak points are more weaker than others, so a quad has an average weakpoint, T5 Naval would have an armored weak point (if at all), and so on and so forth.

Core Units (especially base game) wouldn't have any weak points, since they're not offense units, and T1 units also have no weak points since they are already weak.

Basically, just add at least one weakpoint varying in weakness, and make each of those the critical hitbox.


That would be pretty much the only way to potentially save this suggestion from inevitably getting it's rear handed to by the creator himself.

**Comment by MistahStealYoGril** [comment]
> _**gets flashbacks of TF2**_
> 
> ...Yeah, this is very likely not going to work unless you also implement a way for crits to not be random.
> 
> Let me add to this: **Weak Points**
> 
> Most units have them. Some units don't have a weak point to begin with OR have armored weak points, such as the Oct.
> 
> Take for example: the Corvus. It has a large barrel that looks like a mouth. That would be the weak point, thus critical hits for that specific part of the unit.
> 
> Some weak points are more weaker than others, so a quad has an average weakpoint, T5 Naval would have an armored weak point (if at all), and so on and so forth.
> 
> Core Units (especially base game) wouldn't have any weak points, since they're not offense units, and T1 units also have no weak points since they are already weak.
> 
> Basically, just add at least one weakpoint varying in weakness, and make each of those the critical hitbox.
> 
> 
> That would be pretty much the only way to potentially save this suggestion from inevitably getting it's rear handed to by the creator himself.



That would actually work tho. I was also aiming toward the weapon being disabled if it got hit. I don't know if that is possible.

**Comment by IxGamerXL** [comment]
> > _**gets flashbacks of TF2**_
> > 
> > ...Yeah, this is very likely not going to work unless you also implement a way for crits to not be random.
> > 
> > Let me add to this: **Weak Points**
> > 
> > Most units have them. Some units don't have a weak point to begin with OR have armored weak points, such as the Oct.
> > 
> > Take for example: the Corvus. It has a large barrel that looks like a mouth. That would be the weak point, thus critical hits for that specific part of the unit.
> > 
> > Some weak points are more weaker than others, so a quad has an average weakpoint, T5 Naval would have an armored weak point (if at all), and so on and so forth.
> > 
> > Core Units (especially base game) wouldn't have any weak points, since they're not offense units, and T1 units also have no weak points since they are already weak.
> > 
> > Basically, just add at least one weakpoint varying in weakness, and make each of those the critical hitbox.
> > 
> > 
> > That would be pretty much the only way to potentially save this suggestion from inevitably getting it's rear handed to by the creator himself.
> 
> 
> 
> That would actually work tho. I was also aiming toward the weapon being disabled if it got hit. I don't know if that is possible. 

I am not an advanced programmer on JavaScript, but I do know that this is possible with a _Timer.schedule_ method and changing all of the weapon reload speeds to 0%.

The _Timer.schedue_ method is used so that the weapon regains functionality after a preset amount of time. And with java this likely still works since javascript is just a simplified version of Java.

There could also be a cooldown on the weak point so that the unit doesn't get point camped.

**Comment by MistahStealYoGril** [comment]
> > > _**gets flashbacks of TF2**_
> > > 
> > > ...Yeah, this is very likely not going to work unless you also implement a way for crits to not be random.
> > > 
> > > Let me add to this: **Weak Points**
> > > 
> > > Most units have them. Some units don't have a weak point to begin with OR have armored weak points, such as the Oct.
> > > 
> > > Take for example: the Corvus. It has a large barrel that looks like a mouth. That would be the weak point, thus critical hits for that specific part of the unit.
> > > 
> > > Some weak points are more weaker than others, so a quad has an average weakpoint, T5 Naval would have an armored weak point (if at all), and so on and so forth.
> > > 
> > > Core Units (especially base game) wouldn't have any weak points, since they're not offense units, and T1 units also have no weak points since they are already weak.
> > > 
> > > Basically, just add at least one weakpoint varying in weakness, and make each of those the critical hitbox.
> > > 
> > > 
> > > That would be pretty much the only way to potentially save this suggestion from inevitably getting it's rear handed to by the creator himself.
> > 
> > 
> > 
> > That would actually work tho. I was also aiming toward the weapon being disabled if it got hit. I don't know if that is possible. 
> 
> I am not an advanced programmer on JavaScript, but I do know that this is possible with a _Timer.schedule_ method and changing all of the weapon reload speeds to 0%.
> 
> The _Timer.schedue_ method is used so that the weapon regains functionality after a preset amount of time. And with java this likely still works since javascript is just a simplified version of Java.
> 
> There could also be a cooldown on the weak point so that the unit doesn't get point camped.

I like that idea. Sounds good

**Comment by notrealn** [comment]
Seems unnecessarily complicated and makes t5s much easier to counter.......... plus there is literally no way to make your turrets not shoot towards the target's center, which makes it completely dependent on the enemy ai facing the right way (even with logic). 

Just imagine spending a ton of resources on a corvus, just for a single salvo thats a billion times cheaper compared to the t5 to shut it down completely because it got a single hit when you were trying to shoot it.

**Comment by IxGamerXL** [comment]
> Seems unnecessarily complicated and makes t5s much easier to counter.......... plus there is literally no way to make your turrets not shoot towards the target's center, which makes it completely dependent on the enemy ai facing the right way (even with logic). 
> 
> Just imagine spending a ton of resources on a corvus, just for a single salvo thats a billion times cheaper compared to the t5 to shut it down completely because it got a single hit when you were trying to shoot it. 

Then heres what it'll go like.

Each turret has a certain amount of crit power. Duos and other 1x1 turrets cannot crit.

Turrets like Salvos & Lancers can crit T2 and T3

Bigger turrets like Fuses & Cyclones can crit T2-T4

Only expensive turrets like Foreshadows & Spectres can crit T2-T5 since they both require a devastating amount of resources to build.

Foreshadows have +75% crit power for T2, +50% for T3, and +25% for T4.


Using this set of attributes, this prevents inferior turrets ko'ing tougher units, and makes turret spam less effective and more costly.

**Comment by itcannotbe** [comment]
The thing with crits is that they plainly increase damage by some percentage or some points, not to the point that they can one shot units.

**Comment by notrealn** [comment]
Still seems unnecessarily complicated…. This does not seem that easy to implement and has some balancing issues. Most players would agree that random crits are not fun. This mechanic does not make the game more fun or have more depth. It just adds more complexity to the game for no reason.

Note: how are you supposed to break anti rush walls if this was implemented? most of them use spectres or foreshadows, so trying to use t5s to break them down would be much harder due to them being shut down randomly.

**Comment by MEEPofFaith** [comment]
> _**gets flashbacks of TF2**_

*And this is where I step in and advertise my random crits turret in Progressed Materials.*

**Comment by OnyxFinger** [comment]
> _**gets flashbacks of TF2**_
> 
> ...Yeah, this is very likely not going to work unless you also implement a way for crits to not be random.
> 
> Let me add to this: **Weak Points**
> 
> Most units have them. Some units don't have a weak point to begin with OR have armored weak points, such as the Oct.
> 
> Take for example: the Corvus. It has a large barrel that looks like a mouth. That would be the weak point, thus critical hits for that specific part of the unit.
> 
> Some weak points are more weaker than others, so a quad has an average weakpoint, T5 Naval would have an armored weak point (if at all), and so on and so forth.
> 
> Core Units (especially base game) wouldn't have any weak points, since they're not offense units, and T1 units also have no weak points since they are already weak.
> 
> Basically, just add at least one weakpoint varying in weakness, and make each of those the critical hitbox.
> 
> That would be pretty much the only way to potentially save this suggestion from inevitably getting it's rear handed to by the creator himself.

This wouldn't work either. Most units would either barely be hit on the weak points, or always be hit on the weak points.

**Comment by MistahStealYoGril** [comment]
Discussion seems to be dead so I'll be closing this. Thx for your input everyone.
